src/nvim/api/buffer.c

```diff
@@ -86,6 +86,12 @@ void buffer_del_line(Buffer buffer, Integer index, Error *err)
   FUNC_ATTR_DEFERRED
 {
   Array array = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (index >= buf->b_ml.ml_line_count) {
+    return;
+  }
+
```

Comment:
The following call to `buffer_set_line_slice()` will also check for this (with an additional check for `buf != NULL`). So this code could be removed.